Tests/XMTPTests/ConversationTests.swift

```diff
@@ -0,0 +1,109 @@
+//
+//  ConversationTests.swift
+//
+//
+//  Created by Pat Nakajima on 12/6/22.
+//
+
+import XCTest
+@testable import XMTP
+
+class ConversationTests: XCTestCase {
+	var fakeApiClient: FakeApiClient!
+
+	var alice: PrivateKey!
+	var aliceClient: Client!
+
+	var bob: PrivateKey!
+	var bobClient: Client!
+
+	override func setUp() async throws {
+		alice = try PrivateKey.generate()
+		bob = try PrivateKey.generate()
+
+		fakeApiClient = FakeApiClient()
+
+		aliceClient = try await Client.create(account: alice, apiClient: fakeApiClient)
+		bobClient = try await Client.create(account: bob, apiClient: fakeApiClient)
+	}
+
+	func testCanInitiateV2Conversation() async throws {
+		let existingConversations = try await aliceClient.conversations.list()
+		XCTAssert(existingConversations.isEmpty, "already had conversations somehow")
+
+		guard case let .v2(conversation) = try await bobClient.conversations.newConversation(with: alice.walletAddress) else {
+			XCTFail("Did not get a v2 convo")
+			return
+		}
+
+		let aliceInviteMessage = fakeApiClient.findPublishedEnvelope(.userInvite(alice.walletAddress))
+		let bobInviteMessage = fakeApiClient.findPublishedEnvelope(.userInvite(bob.walletAddress))
+
+		XCTAssert(aliceInviteMessage != nil, "no alice invite message")
+		XCTAssert(bobInviteMessage != nil, "no bob invite message")
+
+		XCTAssertEqual(conversation.peerAddress, alice.walletAddress)
+
+		let newConversations = try await aliceClient.conversations.list()
+		XCTAssertEqual(1, newConversations.count, "already had conversations somehow")
+	}
+
+	func testCanFindExistingV1Conversation() async throws {
+		let encoder = TextCodec()
+		let encodedContent = try encoder.encode(content: "hi alice")
+
+		let date = Date().advanced(by: -1_000_000)
```

Comment:
Similar time q: What's the purpose of it being negative here? I'm reading this as if we're going back in time by 1_000_000 seconds based on [advancedBy / TimeInterval docs](https://developer.apple.com/documentation/foundation/date/3329237-advanced) but idk if that's correct.

Reply:
I just wanted a specific date in the past to test with to make sure it was carried through in the tests, rather than using the current `Date()`, which could lead to things "Just working" even if I was newing up a fresh one somewhere I shouldn't be. I could see how using `1_000_000` would be confusing though so I'll change the magic number and leave a comment to explain what's going on here.

Reply:
Or maybe just name the variable `someTimeAgo` or something.

Reply:
> Or maybe just name the variable someTimeAgo or something.

I like that.